lib/ClangImporter/ImportType.cpp

```diff
@@ -1696,22 +1697,51 @@ ImportedType ClangImporter::Implementation::importFunctionReturnType(
                     OptionalityOfReturn);
 }
 
+static Type
+findGenericTypeInGenericDecls(const clang::TemplateTypeParmType *templateParam,
+                              ArrayRef<GenericTypeParamDecl *> genericParams) {
+  StringRef name = templateParam->getIdentifier()->getName();
+  auto genericParamIter =
+      llvm::find_if(genericParams, [name](GenericTypeParamDecl *generic) {
+        return generic->getName().str() == name;
+      });
+  // TODO: once we support generics in class types, replace this with
```

Comment:
@hlopko Just FYI, I think #33284 will start crashing once this lands. Happy to help resolve the issues if you'd like.

Reply:
No worries, I'll get to that soon :)